indico/web/forms/fields.py

```diff
@@ -540,3 +541,20 @@ def __init__(self, *args, **kwargs):
 
     def _value(self):
         return self.text_value
+
+
+class IndicoPalettePickerField(JSONField):
+    """Field allowing user to pick a color from a set of predefined values"""
+
+    widget = JinjaWidget('forms/palette_picker_widget.html')
+
+    def __init__(self, *args, **kwargs):
+        self.color_list = kwargs.pop('color_list')
+        super(IndicoPalettePickerField, self).__init__(*args, **kwargs)
+
+    def process_formdata(self, valuelist):
+        super(IndicoPalettePickerField, self).process_formdata(valuelist)
+        self.data = ColorTuple(self.data['text'], self.data['background'])
+
+    def populate_obj(self, obj, name):
```

Comment:
What about adding a class-level `CAN_POPULATE` attribute to `JSONField` that's false by default? If set to true in a subclass, `populate_obj` would call super instead of doing nothing.